src/video_canister/src/lib.rs

```diff
@@ -0,0 +1,148 @@
+use ic_cdk::export::candid::{CandidType, Deserialize, Principal,};
+use ic_cdk_macros::{update, query, init, pre_upgrade, post_upgrade};
+use ic_cdk::storage;
+
+#[derive(CandidType, Deserialize)]
+pub struct PutMetaInfo{
+    pub name: String,
+    pub description: String,
+    pub chunk_num: usize,
+}
+
+#[derive(CandidType, Deserialize)]
+pub struct MetaInfo{
+    pub name: String,
+    pub description: String,
+    pub chunk_num: usize,
+    pub version: usize,
+    pub owner: Principal,
+}
+
+impl Default for MetaInfo{
+    fn default() -> Self { 
+        MetaInfo{
+            name: String::new(),
+            description: String::new(),
+            chunk_num: 0,
+            version: CANISTER_VERSION,
+            owner: Principal::anonymous(),
+        }
+     }
+}
+
+#[derive(CandidType, Deserialize)]
+pub enum PutMetaInfoResponse{
+    #[serde(rename = "success")]
+    Success,
+
+    #[serde(rename = "missing_rights")]
+    MissingRights,
+}
+
+#[derive(CandidType, Deserialize)]
+pub enum PutChunkResponse{
+    #[serde(rename = "success")]
+    Success,
+
+    #[serde(rename = "missing_rights")]
+    MissingRights,
+
+    #[serde(rename = "out_of_bounds")]
+    OutOfBounds
+}
+
+#[derive(CandidType, Deserialize)]
+pub enum ChangeOwnerResponse{
+    #[serde(rename = "success")]
+    Success,
+
+    #[serde(rename = "missing_rights")]
+    MissingRights,
+}
+
+type Chunk = Vec<u8>;
+type Chunks = Vec<Chunk>;
+
+const CANISTER_VERSION: usize = 0usize;
```

Comment:
würde diese Zeile weiter oben erwarten, weil in zeile 27 genutzt

Reply:
fixed